src/Bicep.LangServer/Handlers/BicepDeploymentParametersHandler.cs

```diff
@@ -148,7 +148,12 @@ private bool IsExpression(SyntaxBase modifier)
             return modifier is ParameterDefaultValueSyntax parameterDefaultValueSyntax &&
                 parameterDefaultValueSyntax.DefaultValue is ExpressionSyntax expressionSyntax &&
                 expressionSyntax is not null &&
-                expressionSyntax is not StringSyntax &&
+                // Complex Evaluation of StringSyntax is required for nested functions like 'resource${uniqueString(resourceGroup().id)}'
+                // Fixes: https://github.com/Azure/bicep/issues/8154
+                (
+                    expressionSyntax is not StringSyntax ||
```

Comment:
I think this simpler syntax would work:

(expressionSyntax is not StringSyntax || specificationString.IsInterpolated())

(and remove the expressionSyntax is not StringSyntax line)